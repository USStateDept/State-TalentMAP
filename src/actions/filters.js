import axios from 'axios';
import api from '../api';

export function filtersHasErrored(bool) {
  return {
    type: 'FILTERS_HAS_ERRORED',
    hasErrored: bool,
  };
}
export function filtersIsLoading(bool) {
  return {
    type: 'FILTERS_IS_LOADING',
    isLoading: bool,
  };
}
export function filtersFetchDataSuccess(filters) {
  return {
    type: 'FILTERS_FETCH_DATA_SUCCESS',
    filters,
  };
}

export function filtersFetchData(items, queryParams) {
  return (dispatch) => {
    dispatch(filtersIsLoading(true));

    const queryParamObject = queryParams;

    // "mappedParams" allow us to map filters to query params.
    // Since query params come in with no "extra" data, we have
    // to map them back to the original filters from the API
    // in order to supplement them with human-readable data.
    // "filters" will store our selectable filters.
    const responses = { mappedParams: [], filters: [] };

    function dispatchSuccess() {
      // check if we've gotten all the filters we asked for
      if (responses.filters.length === items.filters.length) {
        // check for option queryParamObject to map against (used for pill filters)
        if (queryParamObject) {
          responses.filters.forEach((response) => {
            const filterRef = response.item.selectionRef;
            Object.keys(queryParamObject).forEach((key) => {
              if (key === filterRef) {
                // convert the string to an array
                const paramArray = queryParamObject[key].split(',');
                paramArray.forEach((paramArrayItem) => {
                  // create a base config object
                  const mappedObject = {
                    selectionRef: filterRef,
                    codeRef: paramArrayItem,
                  };
                  responses.filters.forEach((filterItem) => {
                    filterItem.data.forEach((filterItemObject) => {
                      if (filterItemObject.code === mappedObject.codeRef) {
                        // handle boolean filters a little differently
                        // we want to show the title in the pill, not the value
                        if (
                          response.item.title === 'COLA' ||
                          response.item.title === 'Post Differential' ||
                          response.item.title === 'Danger pay' ||
                          response.item.title === 'Domestic'
                        ) {
                          mappedObject.description = response.item.title;
                        } else {
                          // try to get the shortest description since pills should be small
                          mappedObject.description =
                            filterItemObject.short_description ||
                            filterItemObject.description ||
                            filterItemObject.long_description ||
                            filterItemObject.code;
                        }
                      }
                    });
                  });
                  // push our formed object to the mappedParams array
                  responses.mappedParams.push(mappedObject);
                });
              }
            });
          });
        }
        // finally, dispatch a success
        dispatch(filtersIsLoading(false));
        dispatch(filtersFetchDataSuccess(responses));
      }
    }

    items.filters.forEach((item) => {
      // check for filters that don't need to be requested from the API
      if (!item.item.endpoint) {
        responses.filters.push(item);
      } else { // get filters that have an associated endpoint
        axios.get(`${api}/${item.item.endpoint}`)
              .then((response) => {
                const itemFilter = Object.assign({}, item);
                itemFilter.data = response.data.results;
                responses.filters.push({ data: response.data.results, item: itemFilter.item });
                dispatchSuccess();
              })
              .catch(() => dispatch(filtersHasErrored(true)));
      }
    });
  };
}
