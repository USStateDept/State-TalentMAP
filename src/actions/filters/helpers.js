import { isUndefined } from 'lodash';
import { getPostName } from 'utilities';
import { COMMON_PROPERTIES } from 'Constants/EndpointParams';

// Mapping grade codes to a full description for clarity
// If no match is found, return the unmodified code.
export function getCustomGradeDescription(gradeCode) {
  switch (gradeCode) {
    case '00':
      return '00 (Multiple Grades Considered)';
    case 'CM':
      return 'CM Career Minister (FE-CM)';
    case 'MC':
      return 'MC Minister-Counselor (FE-MC)';
    case 'OC':
      return 'OC Counselor (FE-OC)';
    case 'OM':
      return 'Office Manager (OM)';
    default:
      return gradeCode;
  }
}

function getLanguageNameByIfNull(filterItemObject = {}) {
  return filterItemObject.code === COMMON_PROPERTIES.NULL_LANGUAGE ?
    filterItemObject.customDescription || filterItemObject.formal_description
    :
    `${filterItemObject.formal_description} (${filterItemObject.code})`;
}

function getFuncRegionCustomDescription(shortDescription, longDescription) {
  return `(${shortDescription}) ${longDescription}`;
}

function getRegionCustomDescription(shortDescription, longDescription) {
  return `(${shortDescription}) ${longDescription}`;
}

function getSkillCustomDescription(description, code) {
  return `${description} (${code})`;
}

function getCommuterPostDescription(filterItemObject = {}) {
  return `${filterItemObject.description}`;
}

// create a custom description based on the filter type
// eslint-disable-next-line complexity
export function getFilterCustomDescription(filterItem, filterItemObject) {
  const { item: { description: descriptionPrimary } } = filterItem;
  const { short_description: shortDescription, long_description: longDescription, description,
    code, name } = filterItemObject;
  switch (descriptionPrimary) {
    case 'region':
    case 'region-tandem':
      return getRegionCustomDescription(shortDescription, longDescription);
    case 'functionalRegion':
    case 'functionalRegion-tandem':
      return getFuncRegionCustomDescription(shortDescription, longDescription);
    case 'skill':
    case 'skill-tandem':
      return getSkillCustomDescription(description, code);
    case 'post':
      return getPostName(filterItemObject);
    case 'bidCycle':
    case 'bidCycle-tandem':
      return name;
    case 'language':
    case 'language-tandem':
    // language code NONE gets displayed differently
      return getLanguageNameByIfNull(filterItemObject);
    case 'grade':
    case 'grade-tandem':
      return getCustomGradeDescription(code);
    case 'commuterPosts':
      return getCommuterPostDescription(filterItemObject);
    case 'postDiff': case 'dangerPay': case 'bidSeason':
      return description;
    default:
      return false;
  }
}

// create custom attributes based on the filter type
// eslint-disable-next-line complexity
export function getFilterCustomAttributes(filterItem, filterItemObject) {
  const { item: { description: descriptionPrimary } } = filterItem;
  const { code } = filterItemObject;
  switch (descriptionPrimary) {
    case 'language':
    case 'language-tandem':
      if (code === 'NLR') {
        return { group: 'no-language' };
      }
      return { group: 'languages' };
    default:
      break;
  }
  return null;
}

const getDefaultPillText = filterItemObject => (
  filterItemObject.short_description ||
  filterItemObject.description ||
  filterItemObject.long_description ||
  filterItemObject.code ||
  filterItemObject.name ||
  ''
);

// Our standard method for getting a pill description.
// Pass a customType string for special rendering.
export function getPillDescription(filterItemObject, customType) {
  const defaultText = getDefaultPillText(filterItemObject);
  switch (customType) {
    case 'dangerPay':
      return `Danger pay: ${filterItemObject.description}`;
    case 'postDiff':
      return `Post differential: ${filterItemObject.description}`;
    case 'language':
    case 'language-tandem':
      return filterItemObject.custom_description;
    default:
      return defaultText;
  }
}

// when getting pill descriptions for posts or missions, perform alternate method
export function getPostOrMissionDescription(data) {
  if (data.type === 'post') {
    return `${getPostName(data)}`;
  }
  return false;
}

export function doesCodeOrIdMatch(filterItem, filterItemObject, mappedObject) {
  const filterCode = filterItemObject.code;
  const filterRef = filterItem.item.selectionRef;
  const filterId = filterItemObject.id;

  const codeAndRefMatch = !isUndefined(filterCode) &&
    filterCode.toString() === mappedObject.codeRef.toString() &&
    filterRef === mappedObject.selectionRef;

  const idAndRefMatch = !isUndefined(filterId) &&
  filterItemObject.id.toString() === mappedObject.codeRef.toString() &&
  filterRef === mappedObject.selectionRef;

  if (codeAndRefMatch || idAndRefMatch) {
    return true;
  }
  return false;
}

export function isBooleanFilter(description) {
  if (
    description === 'COLA' ||
    description === 'available'
  ) {
    return true;
  }
  return false;
}

export function isPercentageFilter(description) {
  if (
    description === 'dangerPay' ||
    description === 'postDiff'
  ) {
    return true;
  }
  return false;
}
