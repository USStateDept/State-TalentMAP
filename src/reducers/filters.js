const items =
  {
    filters: [
      {
        item: {
          title: 'Skill code',
          sort: 100,
          description: 'skill',
          endpoint: 'skill/',
          selectionRef: 'skill__code__in',
          text: 'Choose skill codes',
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Language',
          sort: 200,
          description: 'language',
          endpoint: 'language/',
          selectionRef: 'languages__language__code__in',
          text: 'Choose languages',
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Grade',
          sort: 300,
          description: 'grade',
          endpoint: 'grade/',
          selectionRef: 'grade__code__in',
          text: 'Choose grades',
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Tour of Duty',
          sort: 400,
          description: 'tod',
          endpoint: 'tour_of_duty/',
          selectionRef: 'post__tour_of_duty__code__in',
          text: 'Choose tour of duty length',
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'Region',
          sort: 500,
          description: 'region',
          endpoint: 'organization/?is_available=true',
          selectionRef: 'organization__code__in',
          text: 'Choose region',
          choices: [
          ],
        },
        data: [
        ],
      },
      {
        item: {
          title: 'COLA',
          sort: 600,
          bool: true, // use bool: true to share a common HTML template
          description: 'COLA',
          selectionRef: 'post__cost_of_living_adjustment__gt',
          text: 'Include only positions with COLA',
          choices: [
          ],
        },
        data: [
          { code: '0', short_description: 'Yes' }, // use a code of 0 to specify we want to return results where COLA > 0
        ],
      },
      {
        item: {
          title: 'Post Differential',
          sort: 700,
          bool: true,
          description: 'postDiff',
          selectionRef: 'post__differential_rate__gt',
          text: 'Include only positions with a post differential',
          choices: [
          ],
        },
        data: [
          { code: '0', short_description: 'Yes' },
        ],
      },
      {
        item: {
          title: 'Danger pay',
          sort: 800,
          bool: true,
          description: 'dangerPay',
          selectionRef: 'post__danger_pay__gt',
          text: 'Include only positions with danger pay',
          choices: [
          ],
        },
        data: [
          { code: '0', short_description: 'Yes' },
        ],
      },
      {
        item: {
          title: 'Domestic',
          sort: 900,
          bool: true,
          description: 'domestic',
          selectionRef: 'is_domestic',
          text: 'Include only domestic positions',
          choices: [
          ],
        },
        data: [
          { code: 'true', short_description: 'Yes' },
        ],
      },
    ],
  };

export function filtersHasErrored(state = false, action) {
  switch (action.type) {
    case 'FILTERS_HAS_ERRORED':
      return action.hasErrored;
    default:
      return state;
  }
}
export function filtersIsLoading(state = false, action) {
  switch (action.type) {
    case 'FILTERS_IS_LOADING':
      return action.isLoading;
    default:
      return state;
  }
}
export function filters(state = items, action) {
  switch (action.type) {
    case 'FILTERS_FETCH_DATA_SUCCESS':
      return action.filters;
    default:
      return state;
  }
}
