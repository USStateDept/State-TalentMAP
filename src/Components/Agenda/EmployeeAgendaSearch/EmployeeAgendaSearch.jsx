/* eslint-disable no-unused-vars */
/* eslint-disable max-len */
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';
import { useEffect, useRef, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Picky from 'react-picky';
import DateRangePicker from '@wojtekmaj/react-daterange-picker';
import { flatten, get, has, isEmpty, sortBy, throttle, uniqBy } from 'lodash';
import FA from 'react-fontawesome';
import { filtersFetchData } from 'actions/filters/filters';
import { agendaEmployeesFetchData, saveAgendaEmployeesSelections } from 'actions/agendaEmployees';
import { bidderPortfolioCDOsFetchData } from 'actions/bidderPortfolio';
import PositionManagerSearch from 'Components/BureauPage/PositionManager/PositionManagerSearch';
import Spinner from 'Components/Spinner';
import TotalResults from 'Components/TotalResults';
import PaginationWrapper from 'Components/PaginationWrapper';
import ExportButton from 'Components/ExportButton';
import SelectForm from 'Components/SelectForm';
import shortid from 'shortid';
import ListItem from 'Components/BidderPortfolio/BidControls/BidCyclePicker/ListItem';
import Alert from 'Components/Alert';
import EmployeeAgendaSearchCard from '../EmployeeAgendaSearchCard/EmployeeAgendaSearchCard';
import EmployeeAgendaSearchRow from '../EmployeeAgendaSearchRow/EmployeeAgendaSearchRow';
import ProfileSectionTitle from '../../ProfileSectionTitle';
import ResultsViewBy from '../../ResultsViewBy/ResultsViewBy';


const EmployeeAgendaSearch = ({ isCDO }) => {
  const childRef = useRef();
  const dispatch = useDispatch();

  const cdos = useSelector(state => state.bidderPortfolioCDOs);
  const cdosIsLoading = useSelector(state => state.bidderPortfolioCDOsIsLoading);
  const filterData = useSelector(state => state.filters);
  const filtersIsLoading = useSelector(state => state.filtersIsLoading);
  const agendaEmployees$ = useSelector(state => state.agendaEmployees);
  const agendaEmployeesIsLoading = useSelector(state => state.agendaEmployeesFetchDataLoading);
  const agendaEmployeesHasErrored = useSelector(state => state.agendaEmployeesFetchDataErrored);
  const userSelections = useSelector(state => state.agendaEmployeesSelections);

  const agendaEmployees = get(agendaEmployees$, 'results', []);

  const isLoading = false; // filtersIsLoading || cdosIsLoading;

  const filters = get(filterData, 'filters', []);

  const hideFilters = true;

  const bureaus = filters.find(f => f.item.description === 'region');
  const bureauOptions = uniqBy(sortBy(get(bureaus, 'data', []), [(b) => b.long_description]), 'long_description');
  const cycles = filters.find(f => f.item.description === 'bidCycle');
  const cycleOptions = uniqBy(sortBy(get(cycles, 'data', []), [(c) => c.custom_description]), 'custom_description');
  const fsbidHandshakeStatus = filters.find(f => f.item.description === 'handshake');
  const fsbidHandshakeStatusOptions = uniqBy(get(fsbidHandshakeStatus, 'data', []), 'code');
  const panels = [new Date(), new Date(), new Date(), new Date(), new Date()];
  // To-Do: Missing org/post data - using location as placeholder
  const posts = filters.find(f => f.item.description === 'post');
  const postOptions = uniqBy(sortBy(get(posts, 'data', []), [(p) => p.city]), 'code');

  // Pagination
  const [page, setPage] = useState(1);
  const [limit, setLimit] = useState(50);
  const [ordering, setOrdering] = useState('Name');
  // Filters
  const [selectedCurrentBureaus, setSelectedCurrentBureaus] = useState([]);
  const [selectedOngoingBureaus, setSelectedOngoingBureaus] = useState([]);
  const [selectedCDOs, setSelectedCDOs] = useState([]);
  // To-Do: Fake creator data
  const [selectedCreators, setSelectedCreators] = useState([]);
  const [selectedCycles, setSelectedCycles] = useState([]);
  const [selectedHandshakeStatus, setSelectedHandshakeStatus] = useState([]);
  const [selectedPanels, setSelectedPanels] = useState([]);
  const [selectedCurrentPosts, setSelectedCurrentPosts] = useState([]);
  const [selectedOngoingPosts, setSelectedOngoingPosts] = useState([]);
  const [selectedTED, setSelectedTED] = useState(null);
  // Free Text
  const [textInput, setTextInput] = useState(get(userSelections, 'textInput', ''));
  const [textSearch, setTextSearch] = useState(get(userSelections, 'textSearch', ''));
  // Controls
  const [cardView, setCardView] = useState(get(userSelections, 'cardView', true));
  const [clearFilters, setClearFilters] = useState(false);

  const count = agendaEmployees ? agendaEmployees.length : 0;

  const view = cardView ? 'card' : 'grid';

  const query = {
    // Pagination
    page,
    limit,
    ordering,
    // User Filters
    // [`${get(bureaus, 'item.selectionRef', '')}-current`]: selectedCurrentBureaus.map(bureauObject => (get(bureauObject, 'code'))),
    // [`${get(bureaus, 'item.selectionRef', '')}-ongoing`]: selectedOngoingBureaus.map(bureauObject => (get(bureauObject, 'code'))),
    // [get(cycles, 'item.selectionRef', '')]: selectedCycles.map(cycleObject => (get(cycleObject, 'id'))),
    // [get(fsbidHandshakeStatus, 'item.selectionRef', '')]: selectedHandshakeStatus.map(fsbidHSStatusObject => (get(fsbidHSStatusObject, 'code'))),
    // selectedPanels,
    // [`${get(posts, 'item.selectionRef', '')}-current`]: selectedCurrentPosts.map(postObject => (get(postObject, 'code'))),
    // [`${get(posts, 'item.selectionRef', '')}-ongoing`]: selectedOngoingPosts.map(postObject => (get(postObject, 'code'))),
    // selectedTED,
    // Free Text
    q: textInput || textSearch,
  };

  const currentInputs = {
    page,
    limit,
    ordering,
    selectedCurrentBureaus,
    selectedOngoingBureaus,
    selectedCDOs,
    selectedCreators,
    selectedCycles,
    selectedHandshakeStatus,
    selectedPanels,
    selectedCurrentPosts,
    selectedOngoingPosts,
    selectedTED,
    textInput,
    textSearch,
    cardView,
  };

  useEffect(() => {
    // dispatch(filtersFetchData(filterData, {}));
    // dispatch(bidderPortfolioCDOsFetchData());
    dispatch(agendaEmployeesFetchData(query));
    dispatch(saveAgendaEmployeesSelections(currentInputs));
  }, []);

  useEffect(() => {
    const filters$ = [
      selectedCurrentBureaus,
      selectedOngoingBureaus,
      selectedCDOs,
      selectedCycles,
      selectedHandshakeStatus,
      selectedPanels,
      selectedCurrentPosts,
      selectedOngoingPosts,
      selectedTED,
    ];
    if (isEmpty(flatten(filters$)) && isEmpty(textSearch)) {
      setClearFilters(false);
    } else {
      setClearFilters(true);
    }
    dispatch(agendaEmployeesFetchData(query));
    dispatch(saveAgendaEmployeesSelections(currentInputs));
  }, [
    page,
    limit,
    ordering,
    selectedCurrentBureaus,
    selectedOngoingBureaus,
    selectedCDOs,
    selectedCycles,
    selectedHandshakeStatus,
    selectedPanels,
    selectedCurrentPosts,
    selectedOngoingPosts,
    selectedTED,
    textSearch,
  ]);

  useEffect(() => {
    dispatch(saveAgendaEmployeesSelections(currentInputs));
  }, [cardView]);

  function submitSearch(text) {
    setTextSearch(text);
  }

  const throttledTextInput = () =>
    throttle(q => setTextInput(q), 300, { leading: false, trailing: true });

  const setTextInputThrottled = (q) => {
    throttledTextInput(q);
  };


  const renderSelectionList = ({ items, selected, ...rest }) => {
    let codeOrID = 'code';
    // only Cycle needs to use 'id'
    if (!has(items[0], 'code')) {
      codeOrID = 'id';
    }
    const getSelected = item => !!selected.find(f => f[codeOrID] === item[codeOrID]);
    let queryProp = 'description';
    if (get(items, '[0].custom_description', false)) queryProp = 'custom_description';
    else if (get(items, '[0].long_description', false)) queryProp = 'long_description';
    else if (get(items, '[0].description', false)) queryProp = 'description';
    else queryProp = 'name';
    return items.map(item =>
      (<ListItem
        key={item[codeOrID]}
        item={item}
        {...rest}
        queryProp={queryProp}
        getIsSelected={getSelected}
      />),
    );
  };

  const pickyProps = {
    numberDisplayed: 2,
    multiple: true,
    includeFilter: true,
    dropdownHeight: 255,
    renderList: renderSelectionList,
    includeSelectAll: true,
  };

  const resetFilters = () => {
    setSelectedCurrentBureaus([]);
    setSelectedOngoingBureaus([]);
    setSelectedCDOs([]);
    setSelectedCycles([]);
    setSelectedHandshakeStatus([]);
    setSelectedPanels([]);
    setSelectedCurrentPosts([]);
    setSelectedOngoingPosts([]);
    setSelectedTED(null);
    setTextSearch('');
    childRef.current.clearText();
    setClearFilters(false);
  };

  const getOverlay = () => {
    let toReturn;
    if (agendaEmployeesIsLoading) {
      toReturn = <Spinner type="bureau-results" class="homepage-position-results" size="big" />;
    } else if (agendaEmployeesHasErrored) {
      toReturn = <Alert type="error" title="Error loading employees" messages={[{ body: 'Please try again.' }]} />;
    } else if (count <= 0) {
      toReturn = <Alert type="info" title="No results found" messages={[{ body: 'Please broaden your search criteria and try again.' }]} />;
    } else {
      toReturn = false;
    }
    if (toReturn) {
      return <div className="usa-width-one-whole empl-search-lower-section results-dropdown">{toReturn}</div>;
    }
    return false;
  };

  const overlay = getOverlay();

  return (
    isLoading ?
      <Spinner type="bureau-filters" size="small" /> :
      <>
        <div className="empl-search-page">
          <div className="usa-grid-full empl-search-upper-section">
            <div className="results-search-bar padded-main-content results-single-search homepage-offset">
              <div className="usa-grid-full results-search-bar-container">
                <ProfileSectionTitle title="Employee Agenda Search" icon="user-circle-o" />
                <PositionManagerSearch
                  submitSearch={submitSearch}
                  onChange={setTextInputThrottled}
                  ref={childRef}
                  textSearch={textSearch}
                  label="Search for an Employee"
                  placeHolder="Search using Employee ID or Name here"
                />
                {
                  !hideFilters &&
                  <>
                    <div className="filterby-container">
                      <div className="filterby-label">Filter by:</div>
                      <div className="filterby-clear">
                        {clearFilters &&
                            <button className="unstyled-button" onClick={resetFilters}>
                              <FA name="times" />
                                  Clear Filters
                            </button>
                        }
                      </div>
                    </div>
                    <div className="usa-width-one-whole empl-search-filters results-dropdown">
                      <div className="filter-div">
                        <div className="label">Cycle:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Select Cycle(s)"
                          value={selectedCycles}
                          options={cycleOptions}
                          onChange={setSelectedCycles}
                          valueKey="id"
                          labelKey="custom_description"
                          disabled
                        />
                      </div>
                      <div className="filter-div">
                        <div className="label">Panel:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Select Panel Date(s)"
                          value={selectedPanels}
                          options={panels}
                          onChange={setSelectedPanels}
                          valueKey="code"
                          labelKey="long_description"
                          disabled
                        />
                      </div>
                      <div className="filter-div split-filter-div">
                        <div className="label">Post:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Current"
                          value={selectedCurrentPosts}
                          options={postOptions}
                          onChange={setSelectedCurrentPosts}
                          valueKey="code"
                          labelKey="custom_description"
                          disabled
                        />
                        <Picky
                          {...pickyProps}
                          placeholder="Ongoing"
                          value={selectedOngoingPosts}
                          options={postOptions}
                          onChange={setSelectedOngoingPosts}
                          valueKey="code"
                          labelKey="custom_description"
                          disabled
                        />
                      </div>
                      <div className="filter-div split-filter-div">
                        <div className="label">Bureau:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Current"
                          value={selectedCurrentBureaus}
                          options={bureauOptions}
                          onChange={setSelectedCurrentBureaus}
                          valueKey="code"
                          labelKey="long_description"
                          disabled
                        />
                        <Picky
                          {...pickyProps}
                          placeholder="Ongoing"
                          value={selectedOngoingBureaus}
                          options={bureauOptions}
                          onChange={setSelectedOngoingBureaus}
                          valueKey="code"
                          labelKey="long_description"
                          disabled
                        />
                      </div>
                      <div className="filter-div handshake-filter-div">
                        <div className="label">Handshake:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Select Handshake Register Status"
                          value={selectedHandshakeStatus}
                          options={fsbidHandshakeStatusOptions}
                          onChange={setSelectedHandshakeStatus}
                          valueKey="code"
                          labelKey="description"
                          disabled
                        />
                      </div>
                      <div className="filter-div">
                        <div className="label">CDO:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Select CDOs"
                          value={selectedCDOs}
                          options={cdos}
                          onChange={setSelectedCDOs}
                          valueKey="id"
                          labelKey="name"
                          disabled
                        />
                      </div>
                      <div className="filter-div">
                        <div className="label">Creator:</div>
                        <Picky
                          {...pickyProps}
                          placeholder="Select Creator(s)"
                          value={selectedCreators}
                          options={cdos}
                          onChange={setSelectedCreators}
                          valueKey="id"
                          labelKey="name"
                          disabled
                        />
                      </div>
                      <div className="filter-div">
                        <div className="label">TED:</div>
                        <DateRangePicker
                          onChange={setSelectedTED}
                          value={selectedTED}
                          maxDetail="month"
                          calendarIcon={null}
                          showLeadingZeros
                          disabled
                        />
                      </div>
                    </div>
                  </>
                }
              </div>
            </div>
          </div>
          {
            !agendaEmployeesIsLoading && !isLoading &&
            <div className="usa-width-one-whole results-dropdown empl-search-controls-container">
              <TotalResults
                total={count}
                pageNumber={1}
                pageSize={count}
                suffix="Results"
                isHidden={isLoading || agendaEmployeesIsLoading}
              />
              <div className="empl-search-controls-right">
                <ResultsViewBy initial={view} onClick={e => setCardView(e === 'card')} />
                {
                  !hideFilters &&
                  <div className="empl-search-results-controls">
                    <SelectForm
                      id="empl-search-num-results"
                      options={[]}
                      label="Sort by:"
                      defaultSort={ordering}
                      onSelectOption={value => setOrdering(value.target.value)}
                      disabled={isLoading}
                    />
                    <SelectForm
                      id="empl-search-num-results"
                      options={[]}
                      label="Results:"
                      defaultSort={limit}
                      onSelectOption={value => setLimit(value.target.value)}
                      disabled={isLoading}
                    />
                  </div>
                }
                <div className="export-button-container">
                  <ExportButton
                    onClick={() => {}}
                    isLoading={isLoading}
                    disabled
                  />
                </div>
              </div>
            </div>
          }
          {
            overlay ||
              <>
                <div className="usa-width-one-whole empl-search-lower-section results-dropdown">
                  {
                    cardView && !agendaEmployeesIsLoading &&
                    <div className="employee-agenda-card">
                      {agendaEmployees.map(emp => (
                        // TODO: include React keys once we have real data
                        <EmployeeAgendaSearchCard key={shortid.generate()} result={emp} isCDO={isCDO} />
                      ))}
                    </div>
                  }
                  {
                    !cardView && !agendaEmployeesIsLoading &&
                    <div className="employee-agenda-row">
                      {agendaEmployees.map(emp => (
                        // TODO: include React keys once we have real data
                        <EmployeeAgendaSearchRow key={shortid.generate()} result={emp} isCDO={isCDO} />
                      ))}
                    </div>
                  }
                </div>
                <div className="usa-grid-full react-paginate empl-search-pagination-controls">
                  {
                    !hideFilters &&
                    <PaginationWrapper
                      pageSize={limit}
                      onPageChange={p => setPage(p.page)}
                      forcePage={page}
                      totalResults={count}
                    />
                  }
                </div>
                {
                  (count >= 50) &&
                  <div className="empl-search-controls-container">
                    <Alert
                      type="info"
                      title="Page Max: 50 results"
                      messages={[{ body: 'Please refine search to see different results.' }]}
                    />
                  </div>
                }
              </>
          }
        </div>
      </>
  );
};

EmployeeAgendaSearch.propTypes = {
  isCDO: PropTypes.bool,
};

EmployeeAgendaSearch.defaultProps = {
  isCDO: false,
};

export default EmployeeAgendaSearch;
