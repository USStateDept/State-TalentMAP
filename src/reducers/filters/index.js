import { filters, filtersHasErrored, filtersIsLoading } from './filters';

export default { filters, filtersHasErrored, filtersIsLoading };
