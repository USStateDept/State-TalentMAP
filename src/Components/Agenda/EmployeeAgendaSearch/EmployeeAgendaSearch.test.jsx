import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import EmployeeAgendaSearch from './EmployeeAgendaSearch';
import filters from '../../../__mocks__/filtersArray';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('EmployeeAgendaSearchComponent', () => {
  it('is defined', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <EmployeeAgendaSearch isCDO />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });

  it('handles loading', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({
        cdos: [],
        cdosIsLoading: true,
        filterData: [],
        filtersIsLoading: true,
      })}
      >
        <MemoryRouter>
          <EmployeeAgendaSearch />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });

  it('handles not loading', () => {
    const wrapper = TestUtils.renderIntoDocument(
      <Provider store={mockStore({
        cdos: [],
        cdosIsLoading: false,
        filterData: filters,
        filtersIsLoading: false,
      })}
      >
        <MemoryRouter>
          <EmployeeAgendaSearch />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot when CDO', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <EmployeeAgendaSearch isCDO />
        </MemoryRouter>
      </Provider>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when not CDO', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <MemoryRouter>
          <EmployeeAgendaSearch isCDO={false} />
        </MemoryRouter>
      </Provider>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
