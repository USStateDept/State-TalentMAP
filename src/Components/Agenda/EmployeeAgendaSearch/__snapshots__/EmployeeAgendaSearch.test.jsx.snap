// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeAgendaSearchComponent matches snapshot when CDO 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <EmployeeAgendaSearch
      isCDO={true}
    />
  </MemoryRouter>
</ContextProvider>
`;

exports[`EmployeeAgendaSearchComponent matches snapshot when not CDO 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <EmployeeAgendaSearch
      isCDO={false}
    />
  </MemoryRouter>
</ContextProvider>
`;
